rundeckapp/grails-app/controllers/rundeck/controllers/FrameworkController.groovy

```diff
@@ -2059,6 +2060,15 @@ class FrameworkController extends ControllerBase implements ApplicationContextAw
 
 
         def bais = new ByteArrayInputStream(params.fileText.toString().getBytes("UTF-8"))
+        def fileValidationResult = nodeSourceFileExceededMaxSize(bais)
+        if( fileValidationResult?.result ){
```

Comment:
This logic should be reversed. Its too conterintuitive that the validation gives "false" if the validation passes. Would be confusing for a future mantainer.

Reply:
WIP